core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -100,6 +101,11 @@ private[spark] class TaskSchedulerImpl(
   // on this class.  Protected by `this`
   private val taskSetsByStageIdAndAttempt = new HashMap[Int, HashMap[Int, TaskSetManager]]
 
+  // keyed by task set stage id
+  // value is true if there have been no resources rejected due to delay scheduling
+  // since the last "full" resource offer
+  private val noDelayScheduleRejects = new mutable.HashMap[Int, Boolean]()
```

Comment:
can we ask each TSM to track it? We can pass the `isAllFreeResources` parameter to TSM when calling `resourceOffer`

Reply:
what do you see as pros/cons of keeping this map vs putting in TSM?
The `TaskSchedulerImpl` is the only one setting/updating the variable based on knowledge only it has (no schedule delay reject).